src/Engine/GameEvents/MoveEvent.java

```diff
@@ -33,14 +36,15 @@ public GameAnimation getEventAnimation(MapView mapView)
   @Override
   public void sendToListener(GameEventListener listener)
   {
-    listener.receiveMoveEvent( this );
+    listener.receiveMoveEvent(this);
   }
 
   @Override
-  public void performEvent(GameMap gameMap)
+  public void performEvent(MapMaster gameMap)
   {
     if( unitPath.getPathLength() > 0 ) // Make sure we have a destination.
     {
+      unitPath.snipCollision(gameMap, unit); // make sure we only go as far as we can go
```

Comment:
For some reason, the animation still animates to the original endpoint, even if the unit bumps into someone along the way. It's not immediately clear why.

Reply:
Oh, of course. Animation happens before execution, so it doesn't know there's something to bump into until after it looks like it didn't happen.

Reply:
Did you want me to resolve this, or did you wanna resolve this, or should it just be an issue?
For now, I kinda like having the animation still fire off the intended location since it lets the game tell you "Yup, you *tried* to go where you wanted to"

Reply:
I got it. I also fixed a problem where units would not update fog after being built. I guess I can push a branch up just so you can see what I'm doing, but I'm not quite done yet.